### ISSUE TITLE
socket.on('disconnect', socket.disconnect) fails

### ISSUE THREAD

**Issue by kengz** [invisibleroads/socketIO-client#107]
The normal behavior for when the server is closed is for the client to close down.
i.e.

```
from socketIO_client import SocketIO
socket = SocketIO('localhost', 8080)
socket.on('disconnect', socket.disconnect)

socket.wait()
```

But when the server is terminated the client keeps trying to reconnect, with the following log:

```
WARNING:root:localhost:8080/socket.io [connection error] recv disconnected (Connection is already closed.)
WARNING:root:localhost:8080/socket.io [connection error] HTTPConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /socket.io/?t=1455316445865-0&transport=polling&EIO=3 (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x10c5c66a0>: Failed to establish a new connection: [Errno 61] Connection refused',))
WARNING:root:localhost:8080/socket.io [connection error] HTTPConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /socket.io/?t=1455316446875-0&transport=polling&EIO=3 (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x10c6d30f0>: Failed to establish a new connection: [Errno 61] Connection refused',))
...
```
##

**Comment by sourabhdesai** [comment]
+1

**Comment by uniquegodwin** [comment]
I need it to reconnect but I need a reconnect event. Please dont remove the reconnection completely. Provide it as an option.

**Comment by sebcagnon** [comment]
I need this as well.
When a client connects, I need to send some basic data. But this is for a local app where the server can be cut and restarted at anytime by user. So I need to know "disconnect" events and/or "reconnect" events.
I get the same warning as Kengz. In this situation it would be really nice to be able to close the connection.

**Comment by invisibleroads** [comment]
Fixed in 4f7c95944d230c68027dcfe2b1c55f7651d631e3 by #136 

@kengz Thank you for pinpointing and reporting the issue.

@uniquegodwin @sebcagnon You can detect the reconnect event either directly or in a namespace. Please see `test_reconnect` and `test_reconnect_with_namespace` for examples. https://github.com/invisibleroads/socketIO-client/blob/master/socketIO_client/tests/__init__.py#L52

    # Directly
    socket.on('connect', on_connect)
    socket.on('disconnect', on_disconnect)
    socket.on('reconnect', on_reconnect)

    # Using namespace
    class Namespace(BaseNamespace):

        def on_connect(self):
            print('[Connected]')

        def on_reconnect(self):
            print('[Reconnected]')

        def on_disconnect(self):
            print('[Disconnected]')

Callbacks defined directly take precedence over callbacks defined in the namespace.
